sdk/cosmos/azure_data_cosmos/src/retry_policies/mod.rs

```diff
@@ -36,33 +43,133 @@ impl RetryResult {
     }
 }
 
-/// Trait defining the retry policy interface for Cosmos DB operations
+/// Enum representing different retry policy strategies for Cosmos DB operations.
+///
+/// This enum provides a type-safe way to handle different retry policies based on the
+/// operation type. Each variant wraps a specific retry policy implementation that handles
+/// particular failure scenarios and retry logic.
+///
+/// # Variants
 ///
-/// This trait provides a contract for implementing retry logic for transient failures
-/// in Azure Cosmos DB operations. Implementers can define custom retry behavior for
-/// both exceptions (errors) and HTTP responses based on their specific requirements.
-#[async_trait]
-pub trait RetryPolicy: Send + Sync {
-    /// Called before sending a request to allow policy-specific modifications
+/// * `Client` - Policy for data plane operations (document CRUD, queries). Handles session
+///   token mismatches, endpoint failures, service unavailability, write forbidden errors,
+///   and resource throttling with multi-region failover support.
+///
+/// * `Metadata` - Policy for metadata operations (database/container management). Handles
+///   transient metadata failures with endpoint failover but simpler retry logic than data
+///   plane operations.
+///
+/// * `ResourceThrottle` - Policy specifically for handling 429 (TooManyRequests) errors
+///   with exponential backoff. Used when rate limiting is the primary concern.
+#[allow(dead_code)]
+pub enum RetryPolicy {
+    /// Data plane retry policy for document operations with comprehensive failure handling
+    Client(Box<ClientRetryPolicy>),
```

Comment:
I wouldn't worry about boxing these types unless we need to. Clippy (the Rust linter we run in CI) has a lint to check for very large enum variants (https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant). There's no need to move things to the heap unless we know they're too big for the stack.

If there's some other reason you're boxing them (or perhaps if Clippy did decide these enum variants are too big), feel free to keep it, but generally the only reasons you need to worry about `Box`ing are if you want a recursive data structure (like a linked list, where a node points to the next node) or if you're dealing with Very Large Data